net/tcp/tcp_send_buffered.c

```diff
@@ -1093,11 +1114,63 @@ ssize_t psock_tcp_send(FAR struct socket *psock, FAR const void *buf,
       psock->s_sndcb->priv  = (FAR void *)psock;
       psock->s_sndcb->event = psock_send_eventhandler;
 
+      /* Allocate a write buffer.  Careful, the network will be momentarily
+       * unlocked here.
+       */
+
+      /* Try to coalesce into the last wrb.
+       *
+       * But only when it might yield larger segments.
+       * (REVISIT: It might make sense to lift this condition.
+       * IOB boundaries and segment boundaries usually do not match.
+       * It makes sense to save the number of IOBs.)
+       *
+       * Also, for simplicity, do it only when we haven't sent anything
+       * from the the wrb yet.
+       */
+
+      const uint32_t max_wrb_size = tcp_max_wrb_size(conn);
```

Comment:
Please move the variable declaration to the beginning of the function!

Reply:
> 
> 
> Please move the variable declaration to the beginning of the function!

Perhaps we should consider adding -c99 to the GCC CFLAGS for compilations in all common directories?

Reply:
> Please move the variable declaration to the beginning of the function!

fixed